### PR TITLE
Colin/confirmation hash

### PR DESCRIPTION
**Pull request by colin-axner**
Client still needs to be updated to reflect changes on the sidechain. 

Addresses issue #76 . Confirmation hash is now `Hash(Hash(msgbytes) + block_hash)`

### CODE REVIEW

File: auth/ante.go [hunk 1]
```diff
@@ -130,15 +133,23 @@ func processConfirmSig(
 	}
 	inputAddresses := plasmaUTXO.GetInputAddresses()
 
-	hash := ethcrypto.Keccak256(signBytes)
+	// Get the block hash
+	blknumKey := make([]byte, binary.MaxVarintLen64)
+	binary.PutUvarint(blknumKey, plasmaUTXO.GetPosition().Get()[0].Uint64())
+	blockHash := metadataMapper.GetMetadata(ctx, blknumKey)
+
+	msgHash := plasmaUTXO.GetMsgHash()
```
Comment (AdityaSripal) [inline auth/ante.go]: Should be `plasmaUTXO.TxHash`

File: docs/overview.md [hunk 2]
```diff
@@ -21,19 +21,19 @@ When the tx bytes are sent to a validator, the validator executes the function V
 
 If ValidateBasic does not return any errors, then the transaction is added to the mempool. CheckTx will then be executed before the tx is included into a block. CheckTx uses the ante handler to verify the authentication data in a transaction, verify the message against the state of the blockchain, and check that the fee provided is sufficient. 
 
-Our ante handler will check that the address that created the signatures in BaseTx match those that own the inputs of the transaction as well as check that the inputs being spent exist. The ante handler also checks that the inputs = outputs + fee. 
+Our ante handler will check that the address that created the signatures in BaseTx match those that own the inputs of the transaction as well as check that the inputs being spent exist. The ante handler also checks that the inputs = outputs + fee. TxConfirmationSigantures will also be checked for each input that is not a deposit nor a fee utxo.
```
Comment (AdityaSripal) [inline docs/overview.md]: Isn't it `InputConfirmationSignatures`?

File: client/plasmacli/cmd/confirmSig.go [hunk 3]
```diff
@@ -1,52 +1,74 @@
...
 	rootCmd.AddCommand(signCmd)
-	signCmd.Flags().String(flagAddr, "", "Address to sign with")
 	viper.BindPFlags(signCmd.Flags())
 }
 
 var signCmd = &cobra.Command{
-	Use:   "sign <position>",
-	Short: "Sign positions to create confirmation signatures, format: 0.0.0.0::0.0.0.0",
-	Args:  cobra.ExactArgs(1),
+	Use:   "sign <position> <address> <signer_address>",
```
[the remark this answers is not in the document]
Comment (AdityaSripal) [reply client/plasmacli/cmd/confirmSig.go]: Use flags to make clearer

File: docs/overview.md [hunk 4]
```diff
@@ -21,19 +21,19 @@ When the tx bytes are sent to a validator, the validator executes the function V
...
 
 Once a transaction has been included into a block, DeliverTx will be executed which will do the same functionality as CheckTx as well as route the Msg to a handler. 
 
-In our implementation, our handler can only handle a SpendMsg. Handling of a SpendMsg includes deleting the inputs of the transaction from our utxo database and creating new utxos corresponding to the addresses and denominations provided in SpendMsg. handleSpendMsg() will utilize the UTXOKeeper which is used to restrict access to our UTXOMapper (utxo database). The handler will also increment the transaction index number so the newly created utxo's will map from the correct position.
+Our implementation utilizes our utxo module in x/utxo. The utxo module allows for modularity when creating a utxo based system. We use our BaseUTXO to implement the UTXO interface. In addition to the base traits of a UTXO, we also keep the MsgHash, TxHash, and InputAddresses in our UTXO. The ProtoUTXO() function allows for our extra information to be stored when the handler in x/utxo creates a new output utxo.
```
Comment (AdityaSripal) [inline docs/overview.md]: Don't keep MsgHash anymore

File: docs/overview.md [hunk 5]
```diff
@@ -21,19 +21,19 @@ When the tx bytes are sent to a validator, the validator executes the function V
...
 
 The UTXOMapper is our utxo database. Our mapper uses keys in the form: < encoded address > + < encoded position > . It maps to the encoded utxo and uses go-amino for its encoding. The < encoded position > at the beginning of the key is used for prefix iteration which will return all the utxo's owned by a specified address. 
 
 ## Types
 
 **SpendMsg**
 
-A SpendMsg contains the position of the input utxos (block number, transaction index, output index, deposit number), confirm signatures signed by the owners of the parent inputs (the inputs to the SpendMsg inputs), the addresses of the input owners, the addresses of the outputs, the denomination of each output, and the fee amount. 
+A SpendMsg contains the position of the input utxos (block number, transaction index, output index, deposit number), transaction confirmation signatures signed by the owners of the parent inputs (the inputs to the SpendMsg inputs), the addresses of the outputs, the amount of each output, and the fee amount.
```
Comment (AdityaSripal) [inline docs/overview.md]: input confirmation signatures not transaction confirmation signatures

File: client/plasmacli/cmd/confirmSig.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (colin-axner) [reply client/plasmacli/cmd/confirmSig.go]: updated

File: docs/overview.md (hunk 2 above)
Comment (colin-axner) [reply docs/overview.md]: done

File: docs/overview.md (hunk 4 above)
Comment (colin-axner) [reply docs/overview.md]: done

File: docs/overview.md (hunk 5 above)
Comment (colin-axner) [reply docs/overview.md]: fixed

**Pull request merged**